lib/irb/color.rb

```diff
@@ -128,10 +128,13 @@ def colorize_code(code, complete: true, ignore_error: false, colorable: colorabl
 
         symbol_state = SymbolState.new
         colored = +''
-        length = 0
-        end_seen = false
 
         scan(code, allow_last_error: !complete) do |token, str, expr|
+          if token.nil?
```

Comment:
When I commented out this entire condition, the tests still passed. 
I think it's because the `str` will still be passed to `Reline::Unicode.escape_for_print` at line `145` and then be pushed to `colored` at line `150`. So the end result is still the same.

Therefore, this condition doesn't seem to be needed?

Reply:
It actually works without this change, (I did not notice that) but I think this code is necessary.

I think `symbol_state.scan_token(token)` and `dispatch_seq` is originally designed to receive `Ripper::Lexer::Elem#event` that is not nil.
The current implementation of scan_token and dispatch_seq works just by chance with nil, but I don't think we should pass nil to it.

Reply:
I see, thanks for the clear explanation 👍